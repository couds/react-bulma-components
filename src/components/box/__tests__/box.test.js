import PropTypes from 'prop-types';
import React from 'react';
import { render } from 'react-testing-library';
import { Box } from '..';

describe('Box component', () => {
  it('Should have box classname', () => {
    const { asFragment } = render(<Box>Facebook</Box>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const { asFragment } = render(<Box className="other-class test">Facebook</Box>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should render as an html section', () => {
    const { asFragment } = render(<Box renderAs="section">This should be a section</Box>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const { asFragment } = render(
      <Box renderAs="section" style={{ width: 200, zIndex: 1 }}>
        This should be a section with custom styles
      </Box>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should accept a react Element as renderAs prop', () => {
    // eslint-disable-next-line react/prop-types
    const Custom = props => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );

    Custom.propTypes = { children: PropTypes.node.isRequired };

    const { asFragment } = render(<Box renderAs={Custom}>This should be a p element</Box>);
    expect(asFragment()).toMatchSnapshot();
  });
});
