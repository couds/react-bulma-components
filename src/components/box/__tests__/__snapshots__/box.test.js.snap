// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should accept a react Element as renderAs prop 1`] = `
<DocumentFragment>
  <p
    class="box"
  >
    CustomThis should be a p element
  </p>
</DocumentFragment>
`;

exports[`Box component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="box other-class test"
  >
    Facebook
  </div>
</DocumentFragment>
`;

exports[`Box component should have custom inline styles 1`] = `
<DocumentFragment>
  <section
    class="box"
    style="width: 200px; z-index: 1;"
  >
    This should be a section with custom styles
  </section>
</DocumentFragment>
`;

exports[`Box component should render as an html section 1`] = `
<DocumentFragment>
  <section
    class="box"
  >
    This should be a section
  </section>
</DocumentFragment>
`;

exports[`Box component should render 1`] = `
<DocumentFragment>
  <div
    class="box"
  />
</DocumentFragment>
`;
